src/adapters/http/server.ts

```diff
@@ -0,0 +1,131 @@
+import Adapter from "../../lib/adapter.js"
+import GleeMessage from "../../lib/message.js"
+import http from "http"
+import { validateData } from "../../lib/util.js"
+import GleeError from "../../errors/glee-error.js"
+import * as url from "url"
+
+class HttpAdapter extends Adapter {
+  private httpResponses = new Map()
+
+  name(): string {
+    return "HTTP server"
+  }
+
+  async connect(): Promise<this> {
+    return this._connect()
+  }
+
+  async send(message: GleeMessage): Promise<void> {
+    return this._send(message)
+  }
+
+  async _connect(): Promise<this> { // NOSONAR
+    const config = await this.resolveProtocolConfig("http")
+    const httpOptions = config?.server
+    const serverUrl = new URL(this.serverUrlExpanded)
+    const httpServer = httpOptions?.httpServer || http.createServer()
+    const asyncapiServerPort = serverUrl.port || 80
+    const optionsPort = httpOptions?.port
+    const port = optionsPort || asyncapiServerPort
+
+    httpServer.on("request", (req, res) => {
+      res.setHeader("Content-Type", "application/json")
+      const bodyBuffer = []
+      let body: object
+      req.on("data", (chunk) => {
+        bodyBuffer.push(chunk)
+      })
+      req.on("end", () => {
+        body = JSON.parse(Buffer.concat(bodyBuffer).toString())
+        this.httpResponses.set(this.serverName, res)
+        let { pathname } = new URL(req.url, serverUrl)
+        pathname = pathname.startsWith("/") ? pathname.substring(1) : pathname
+        if (!this.parsedAsyncAPI.channel(pathname)) {
+          res.end("HTTP/1.1 404 Not Found1\r\n\r\n")
+          const err = new Error(
+            `A client attempted to connect to channel ${pathname} but this channel is not defined in your AsyncAPI file. here`
+          )
+          this.emit("error", err)
+          return err
+        }
+        const { query: query } = url.parse(req.url, true)
```

Comment:
Wouldn't it be enough like this?

```suggestion
        const { query } = url.parse(req.url, true)
```